### PR TITLE
Support CUDA 11.5.1

### PR DESCRIPTION
**Pull request by takagi**

### CODE REVIEW

**Comment by takagi** [comment]
FlexCI Dockerfiles are also needed to update for 11.5.1.

File: install/cupy_builder/cupy_setup_build.py [hunk 1]
```diff
@@ -575,30 +575,34 @@ def _nvcc_gencode_options(cuda_version):
...
                          'compute_50',
                          ('compute_60', 'sm_60'),
                          ('compute_61', 'sm_61'),
                          ('compute_70', 'sm_70'),
                          ('compute_75', 'sm_75'),
-                         'compute_70']
-        elif cuda_version >= 9020:
+                         ('compute_80', 'sm_80'),
+                         'compute_80']
+        elif cuda_version >= 10000:
```
Comment (kmaehashi) [inline install/cupy_builder/cupy_setup_build.py:599]:
```suggestion
        elif cuda_version >= 10020:
```

c.f. https://docs.nvidia.com/cuda/archive/10.2/cuda-compiler-driver-nvcc/index.html#virtual-architecture-feature-list

**Review by kmaehashi — CHANGES_REQUESTED** [review]

**Comment by leofang** [comment]
Let me do a test drive 😅

/test mini

**Comment by takagi** [comment]
PTAL!

**Comment by leofang** [comment]
FYI: https://github.com/cupy/cupy/pull/6296#issuecomment-1007689054

File: .pfnci/schema.yaml [hunk 2]
```diff
@@ -46,7 +46,7 @@ cuda:
   "11.4":
     full_version: "11.4.0"
   "11.5":
-    full_version: "11.5.0"
+    full_version: "11.5.1"
```
Comment (kmaehashi) [inline .pfnci/schema.yaml:49]: Let's pin CI to 11.5.0 as 11.5.1 seems to have bugs in cuSPARSE.

```suggestion
    full_version: "11.5.0"
```

**Review by kmaehashi — CHANGES_REQUESTED** [review]

File: .pfnci/schema.yaml (hunk 2 above)
Comment (takagi) [reply .pfnci/schema.yaml]: I'll regenerate the CI files.

**Review by kmaehashi — APPROVED** [review]

**Pull request merged**